lib/new_relic/agent/custom_event_aggregator.rb

```diff
@@ -70,15 +72,20 @@ def create_custom_event_attributes(attributes)
             key = key[0, MAX_NAME_SIZE]
           end
 
-          # value is limited to 4095
+          # value is limited to 4095 expect for LLM content-related events
           if val.is_a?(String) && val.length > MAX_ATTRIBUTE_SIZE
-            val = val[0, MAX_ATTRIBUTE_SIZE]
+            val = val[0, MAX_ATTRIBUTE_SIZE] unless llm_exempt_event_attribute?(type, key)
           end
 
           new_result[key] = val
         end
       end
 
+      def llm_exempt_event_attribute?(type, key)
```

Comment:
Just to confirm... are we certain that `type` and `key` will always both be strings 100% of the time?

Reply:
For type, we do some logic about type earlier in the process that would have already required it to be a string. 
For key, since its a hash key it should only really be string or symbols anyways. plus it wouldn't check the key until after it matches type, which means we're the ones who have defined key, which they are all strings.

Reply:
Thanks Tanna!